arwen/host/execution.go

```diff
@@ -691,16 +693,20 @@ func (host *vmHost) RevertESDTTransfer(input *vmcommon.ContractCallInput) {
 	vmOutput, err := host.blockChainHook.ProcessBuiltInFunction(revertInput)
 	if err != nil {
 		log.Error("RevertESDTTransfer failed", "error", err)
+		host.meteringContext.UseGas(host.meteringContext.GasLeft())
+		host.runtimeContext.FailExecution(err)
 		return
 	}
 	if vmOutput != nil && vmOutput.ReturnCode != vmcommon.Ok {
 		log.Error("RevertESDTTransfer failed", "returnCode", vmOutput.ReturnCode, "returnMessage", vmOutput.ReturnMessage)
+		host.meteringContext.UseGas(host.meteringContext.GasLeft())
+		host.runtimeContext.FailExecution(errors.New(vmOutput.ReturnMessage))
 	}
 }
 
 // ExecuteESDTTransfer calls the process built in function with the given transfer for ESDT/ESDTNFT if nonce > 0
 // there are no NFTs with nonce == 0
-func (host *vmHost) ExecuteESDTTransfer(destination []byte, sender []byte, tokenIdentifier []byte, nonce uint64, value *big.Int, callType vmcommon.CallType) (*vmcommon.VMOutput, uint64, error) {
+func (host *vmHost) ExecuteESDTTransfer(destination []byte, sender []byte, tokenIdentifier []byte, nonce uint64, value *big.Int, callType vmcommon.CallType, isRevert bool) (*vmcommon.VMOutput, uint64, error) {
```

Comment:
Instead of using a bool variable, we could have had 2 functions for this.

Reply:
2 functions with copy pasted code ?

Reply:
2 functions with extracted duplicated code

Reply:
can leave as it is for now